### ISSUE TITLE
Neural Voice Cloning with a Few Samples

### ISSUE THREAD

**Issue by flrngel** [flrngel/understanding-ai#8]
https://arxiv.org/abs/1802.06006
Paper from Baidu Research

**Comment by flrngel** [comment]
# Abstract
Paper will do
- Speaker adaption
	- fine-tuning a multi-speaker generative model
- Speaker encoding
	- infer speaker embedding which will be used with a multi-speaker generative model

# 1. Introduction
- Text carries linguistic information
- Speaker representation captures speaker's characteristics (pitch, speech rate, accent)
- This paper focuses on voice cloning
- Compares speech naturalness, speaker similarity, cloning/inference time, model footprint

# 2. Voice Cloning

![image](https://user-images.githubusercontent.com/2807595/36942246-1bc6c7b4-1fb1-11e8-9116-467c6e6b9c49.png)

**Paper Notations**
- **f**: multi-speaker generative model
- **g**: speaker encoding function
- **t**: text
- **s**: speaker
- **a**: audio
- **S**: speaker set
- **A**: audio set

## 2.1. Speaker adaption
### Speaker adaption function
![image](https://user-images.githubusercontent.com/2807595/36942213-741c5074-1fb0-11e8-98e8-aac75b777e15.png)

## 2.2. Speaker encoding
### Speaker encoding function
![image](https://user-images.githubusercontent.com/2807595/36942221-86511b12-1fb0-11e8-85c1-531c95841d4c.png)
Paper avoids ***mode collapse*** with training **speaker encoder** seperately

### Loss function (L1)
![image](https://user-images.githubusercontent.com/2807595/36942241-f8f9470c-1fb0-11e8-9955-3498558d1366.png)

### Architecture
![image](https://user-images.githubusercontent.com/2807595/36942255-61620086-1fb1-11e8-965d-bc3ea808256f.png)
- Spectral processing
- Temporal processing
- Cloning sample attention
	- uses multi-head self-attention from ***Transformer***

## 2.3. Discriminative models for evaluation
Because human is so expensive, paper propose those two solutions for evaluation
### 2.3.1. Speaker Classification
- Put additional embedding layer before softmax function from whole architecture

### 2.3.2. Speaker Verification
- binary classification wheter the test audio and enrolled audio are same speaker
![image](https://user-images.githubusercontent.com/2807595/36942278-ee718c1c-1fb1-11e8-8afc-58c345157d30.png)

# Experiments
## 3.1. Datasets
- ***LibriSpeech*** dataset for multi-speaker generative model & speaker encoder model
- sampling from ***VCTK*** for voice cloning
